network/libp2p_impl.go

```diff
@@ -341,27 +327,12 @@ func (impl *libp2pDataTransferNetwork) Protocol(ctx context.Context, id peer.ID)
 	return s.Protocol(), nil
 }
 
-func (impl *libp2pDataTransferNetwork) setPeerProtocol(p peer.ID, proto protocol.ID) {
-	impl.pplk.Lock()
-	defer impl.pplk.Unlock()
+func (impl *libp2pDataTransferNetwork) setDataTransferProtocols(protocols []protocol.ID) {
+	impl.dtProtocols = append([]protocol.ID{}, protocols...)
 
-	impl.peerProtocols[p] = proto
-}
-
-func (impl *libp2pDataTransferNetwork) clearPeerProtocol(p peer.ID) {
-	impl.pplk.Lock()
-	defer impl.pplk.Unlock()
-
-	delete(impl.peerProtocols, p)
-}
-
-// Clear the peer protocol version cache for the peer when the peer disconnects
-func (impl *libp2pDataTransferNetwork) Disconnected(n network.Network, conn network.Conn) {
-	impl.clearPeerProtocol(conn.RemotePeer())
+	// Keep a string version of the protocols for performance reasons
+	impl.dtProtocolStrings = make([]string, 0, len(impl.dtProtocols))
```

Comment:
@dirkmc Why do we need this now ?

Reply:
Because `FirstSupportedProtocol(id, protos...)` takes an array of strings (not an array of `protocol.ID`)